l1-contracts/contracts/state-transition/chain-deps/facets/Executor.sol

```diff
@@ -243,8 +260,82 @@ contract ExecutorFacet is ZKChainBase, IExecutor {
+                (viaIR.logSender2, ) = UnsafeBytes.readAddress(emittedL2Logs, i + L2_LOG_ADDRESS_OFFSET);
+                // slither-disable-next-line unused-return
+                (viaIR.logKey2, ) = UnsafeBytes.readUint256(emittedL2Logs, i + L2_LOG_KEY_OFFSET);
+                // slither-disable-next-line unused-return
+                (viaIR.logValue2, ) = UnsafeBytes.readBytes32(emittedL2Logs, i + L2_LOG_VALUE_OFFSET);
+                i = i.uncheckedAdd(L2_TO_L1_LOG_SERIALIZE_SIZE);
+                // slither-disable-next-line unused-return
+                (viaIR.logSender3, ) = UnsafeBytes.readAddress(emittedL2Logs, i + L2_LOG_ADDRESS_OFFSET);
+                // slither-disable-next-line unused-return
+                (viaIR.logKey3, ) = UnsafeBytes.readUint256(emittedL2Logs, i + L2_LOG_KEY_OFFSET);
+                // slither-disable-next-line unused-return
+                (viaIR.logValue3, ) = UnsafeBytes.readBytes32(emittedL2Logs, i + L2_LOG_VALUE_OFFSET);
+                i = i.uncheckedAdd(L2_TO_L1_LOG_SERIALIZE_SIZE);
+                // slither-disable-next-line unused-return
+                (viaIR.logSender4, ) = UnsafeBytes.readAddress(emittedL2Logs, i + L2_LOG_ADDRESS_OFFSET);
+                // slither-disable-next-line unused-return
+                (viaIR.logKey4, ) = UnsafeBytes.readUint256(emittedL2Logs, i + L2_LOG_KEY_OFFSET);
+                // slither-disable-next-line unused-return
+                (viaIR.logValue4, ) = UnsafeBytes.readBytes32(emittedL2Logs, i + L2_LOG_VALUE_OFFSET);
+                if (logSender != L2_BOOTLOADER_ADDRESS) {
+                    revert InvalidLogSender(logSender, logKey);
+                }
+                if (viaIR.logSender2 != L2_BOOTLOADER_ADDRESS) {
+                    revert InvalidLogSender(viaIR.logSender2, viaIR.logKey2);
+                }
+                if (viaIR.logSender3 != L2_BOOTLOADER_ADDRESS) {
+                    revert InvalidLogSender(viaIR.logSender3, viaIR.logKey3);
+                }
+                if (viaIR.logSender4 != L2_BOOTLOADER_ADDRESS) {
+                    revert InvalidLogSender(viaIR.logSender4, viaIR.logKey4);
+                }
+                if (viaIR.logKey2 != logKey.uncheckedAdd(1)) {
+                    revert InvalidLogKey(logKey.uncheckedAdd(1), viaIR.logKey2);
+                }
+                if (viaIR.logKey3 != logKey.uncheckedAdd(2)) {
+                    revert InvalidLogKey(logKey.uncheckedAdd(2), viaIR.logKey3);
+                }
+                if (viaIR.logKey4 != logKey.uncheckedAdd(3)) {
+                    revert InvalidLogKey(logKey.uncheckedAdd(3), viaIR.logKey4);
+                }
+                if (uint256(logValue) == block.chainid) {
+                    IMessageRoot messageRootContract = IBridgehub(s.bridgehub).messageRoot();
+                    bytes32 historicalRoot = messageRootContract.historicalRoot(uint256(viaIR.logValue2));
```

Comment:
definitely need to name the fields in the `viaIR` struct (or use constants for indeces). Otherwise it is a bit hard to understand what does `viaIR.logValue2` stand for